### PR TITLE
#172899987 Allow access depending on roles

### PR DESCRIPTION
**Pull request by AbdoulNuru**
#### What does this PR do?
- display pages to users according to their roles.
#### Description of Task to be completed?
- only show users pages that their privileges allow them to access.
#### How should this be manually tested?
- clone the repo, then checkout to this branch ``ft-user-access-rights-172899987``
- run `npm i` to install all project dependencies
- run `npm run dev` to start the development server
- login using credentials below for different roles.
```
> super administrator
email: abdoulniyigena@gmail.com
password: password

> travel administrator
email: octopusbn@gmail.com
password: password

> manager
email: needs.grid@gmail.com
password: password
```
#### What are the relevant pivotal tracker stories?
[#172899987](https://www.pivotaltracker.com/story/show/172899987)
#### Screenshots (if appropriate)
N/A

### CODE REVIEW

File: src/components/DrawerComponents.js [hunk 1]
```diff
@@ -97,44 +97,117 @@ const useStyles = makeStyles((theme) => ({
...
+      },
+
+      {
+        key: "Profile",
+        text: "My Profile",
+        to: "/profile-settings",
+        icon: <PersonIcon />,
+      },
+    ];
+  }
```
Comment (AJAkimana) [inline src/components/DrawerComponents.js]: The code is redundant, make use of **destructuring assignment**

**Review by AJAkimana — CHANGES_REQUESTED** [review]
Good work for the approach you've come up with. However, it has a drawback. Even if the routes are hidden from a user, they are still accessible via URL. You need another approach that would redirect the user where they belong if they try to manually type the URL.

File: src/components/DrawerComponents.js (hunk 1 above)
Comment (AbdoulNuru) [reply src/components/DrawerComponents.js]: Thanks, I have used the destructuring way.

**Comment by AbdoulNuru** [comment]
> Good work for the approach you've come up with. However, it has a drawback. Even if the routes are hidden from a user, they are still accessible via URL. You need another approach that would redirect the user where they belong if they try to manually type the URL.

Thanks, changes were made.

**Review by hakoemmy — APPROVED** [review]
@AbdoulNuru Well done, keep up the good work  🙌🙌🙌🙌🙌🙌🙌🙌🙌🙌

**Review by blaise82 — APPROVED** [review]

**Review by Moise1 — APPROVED** [review]

**Review by AJAkimana — APPROVED** [review]

**Pull request merged**